src/characters/universal_bob.ts

```diff
@@ -46,9 +47,57 @@ export class tDecDecrypter {
     return this.keyring.decrypt(messageKit);
   }
 
+  private async authWithSignature(
+    web3Provider: Web3Provider,
+    messageKits: readonly MessageKit[],
+    condition: ConditionSet,
+  ): Promise<string> {
+
+    let mkBytes = new Uint8Array();
+    for (const mk of messageKits) {
+      mkBytes = Buffer.concat([mkBytes, mk.toBytes()]);
+    }
+    const mkHash = sha256(mkBytes);
+
+    const salt = ethersUtils.randomBytes(32);
+    const typedData = {
+      types: {
+        EIP712Domain: [
+          { name: "name", type: "string" },
+          { name: "version", type: "string" },
+          { name: "chainId", type: "uint256" },
+          { name: "salt", type: "bytes32" },
+        ],
+        Condition: [
+          { name: "address", type: "address" },
+          { name: "condition", type: "string" },
+          { name: "messageKitsHash", type: "bytes32" },
+        ]
+      },
+      domain: {
+        name: 'tDec',
+        version: '1',
+        chainId: 1,
+        salt,
+      },
+      message: {
+        address: web3Provider.signer._address,
+        condition: condition.toJSON(),
+        messageKitsHash: mkHash,
+      }
+    }
+
+    return web3Provider.signer._signTypedData(typedData.domain, typedData.types, typedData.message);
+  }
+
   public async retrieveAndDecrypt(
-    messageKits: readonly MessageKit[]
+    messageKits: readonly MessageKit[],
+    web3Provider: Web3Provider,
+    condition: ConditionSet,
   ): Promise<readonly Uint8Array[]> {
+    // TODO: What do I do with this signature?
```

Comment:
Excellent entry point here for `ReencryptionRequest.context` (add a field there).